sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/AggregationIterator.scala

```diff
@@ -321,7 +321,7 @@ abstract class AggregationIterator(
   // Initializing the function used to generate the output row.
   protected val generateOutput: (InternalRow, MutableRow) => InternalRow = {
     val rowToBeEvaluated = new JoinedRow
-    val safeOutputRow = new GenericMutableRow(resultExpressions.length)
+    val safeOutputRow = new SpecificMutableRow(resultExpressions.map(_.dataType))
```

Comment:
These changes in `AggregationIterator.scala` are just some minor optimizations motivated by things that I noticed while working on the Tungsten version. Here, and down below, I noticed that we might be unnecessarily boxing via our use of GenericMutableRow, hence this change.